[WIP] Rebase to upstream v0.8.0

alebedev87 opened the pull request.

seanmalloy commented:
I believe the OWNERS file needs to be updated before this can be merged. @jmelis and @csrwng do you have any suggestions as to who should be in the OWNERS file?

seanmalloy commented:
I'm also pretty certain there will be some OpenShift specific CI/CD configs that will need to be modified as part of this PR. I'll try to investigate what changes are needed, but I'm not familiar with the CI/CD tools setup for OpenShift. Any suggestions to point me in the right directly would be greatly appreciated.

seanmalloy commented:
@abutcher @csrwng @dgoodwin @joelddiaz @twiest I added you to the `OWNERS` file in commit https://github.com/openshift/external-dns/pull/7/commits/5471d3f71684b2cdef04a122d8a0a7783a467071. Let me know if I do not have the correct list of people listed as approvers. Thanks!

seanmalloy commented:
/assign @csrwng

csrwng commented:
Hi @seanmalloy when we started this repo, the thought was that the master branch would remain exactly the same as upstream, while the `openshift-4.0` branch would contain our openshift-specific modifications. It doesn't have to continue that way though so I'll leave it up to you.
This repo was used by the Hive team to manage DNS records for DNS Zone linking. However, we've since transitioned from it and are simply handling it our controllers. Honestly, I don't think anybody in the Hive team needs to be an owner anymore. 
If you have a purpose for this repository going forward, then I'd be all for transferring ownership to you.
/cc @dgoodwin

seanmalloy commented:
@csrwng I'm trying to get k8s external-dns added to OpenShift as a new feature. I have an RFE opened through the official RedHat support portal. I'm willing to help contribute code to this effort.

It looks like this repo was not setup as a fork of the upstream repo when it was originally created. Is it possible to delete this repo and fork it from the upstream git repo, https://github.com/kubernetes-sigs/external-dns, into the openshift org?

After it is forked I'd then update the OWNERS file and add the OpenShift specific CI bits(Dockerfiles, etc). I'm fine being added to the OWNERS file, but my guess is I would need to be a member of the openshift GitHub org first. What is the process to get added to the org?

csrwng commented:
@seanmalloy Let's start by recreating the repo as a fork. Are you ok with this repo getting deleted now? I'll submit a ticket to get it recreated when you tell me so.

seanmalloy commented:
@csrwng I'm ready. You can delete this repo and re-create it as a fork. Thanks for your help!

sgreene570 committed:
UPSTREAM: <carry>: openshift: Add DOWNSTREAM_OWNERS

This commit adds a DOWNSTREAM_OWNERS file to reflect
OpenShift-appropriate Reviewers and Approvers.

alebedev87 committed:
UPSTREAM: <carry>: openshift: OpenShift dockerfiles added

alebedev87 committed:
UPSTREAM: <carry>: openshift: Add .ci-operator.yaml with base_root_image

alebedev87 committed:
UPSTREAM: <carry>: openshift: Update vendor after sync to v0.8.0
